tools/AzureMarkdownRewriterTool/Program.cs

```diff
@@ -20,11 +20,13 @@ namespace Microsoft.DocAsCode.Tools.AzureMarkdownRewriterTool
 
     internal sealed class Program
     {
-        private static readonly Regex _azureHtmlIncludeRegex = new Regex(@"^(\<br\s*\/\>)(\s*\r?\n\[AZURE\.INCLUDE)", RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.Multiline);
+        private static readonly Regex _azureHtmlIncludeWithPrefixRegex = new Regex(@"^(\<br\s*\/\>)(\s*\r?\n\[AZURE\.INCLUDE)", RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.Multiline);
 
-        public static readonly Regex _azureHtmlDefinitionRegex = new Regex(@"^( +)(\[([^\]]+)\]: *<?([^\s>]+)>?(?: +[""(]([^\n]+)["")])? *(?:\r?\n+|$))", RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.Multiline);
+        private static readonly Regex _azureHtmlIncludeWithPostfixRegex = new Regex(@"^(\[AZURE\.INCLUDE\s*\[((?:\[[^\]]*\]|[^\[\]]|\](?=[^\[]*\]))*)\]\(\s*<?([^)]*?)>?(?:\s+(['""])([\s\S]*?)\3)?\s*\)\])[\t\f ]*(\S.*)$", RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.Multiline);
 
-        public static IReadOnlyList<string> SystemMarkdownFileName = new List<string> { "TOC.md" };
+        private static readonly Regex _azureHtmlDefinitionWithLeadingWhitespacesRegex = new Regex(@"^( +)(\[([^\]]+)\]: *<?([^\s>]+)>?(?: +[""(]([^\n]+)["")])? *(?:\r?\n+|$))", RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.Multiline);
+
+        private static IReadOnlyList<string> SystemMarkdownFileName = new List<string> { "TOC.md" };
```

Comment:
I have to say, the naming rule is a little strange, for private static w/o readonly